tests/test_deploy.py

```diff
@@ -89,3 +91,25 @@ def test_rsync_command_username_in_url(tmpdir, mocker, env):
         str(output_path) + '/',
         'fakeuser@example.com:/'
     ],)
+
+
+def test_publish_target_url_error_handling(tmpdir, mocker, env):
+    output_path = tmpdir.mkdir("output")
+    target_url = None
+    publisher = GithubPagesPublisher(env, str(output_path))
+    
+    with pytest.raises(Exception) as exception_info:
+        mocked_publish = mocker.patch("lektor.publisher.publish")   
+        raise Exception('PublishError: Publishing target does not have a host name: ')
+    assert 'Publishing target does not have a host name: ' in str(exception_info._excinfo[1]) 
+
+
+def test_publisher_error_handling(tmpdir, mocker, env):
+    output_path = tmpdir.mkdir("output")
+    target_url = url_parse("ghpages+https://pybee/pybee.github.io?cname=pybee.org")
+    publisher = None
+    mocked_publish = mocker.patch("lektor.publisher.publish")
+    with pytest.raises(Exception) as exception_info:   
+        raise Exception('Server "%s" is not configured for a valid publishing method')
```

Comment:
Same problem here: the test isn't actually testing the code you've modified. For this, I would suggest something like this:

```python
def test_publish_bad_scheme(tmpdir, env):
    output_path = tmpdir.mkdir("output")
    target_url = "badscheme://host/path"

    with pytest.raises(PublishError) as excinfo:
        publish(env, target_url, output_path)

    expected_message = (
        "Server "host" is not configured for a valid publishing method: "
        "badscheme is an unknown scheme."
    )
    assert str(excinfo.value) == expected_message
````